sound/soc/sof/topology.c

```diff
@@ -146,11 +148,20 @@ static int sof_keyword_dapm_event(struct snd_soc_dapm_widget *w,
 	dev_dbg(sdev->dev, "received event %d for widget %s\n",
 		event, w->name);
 
+	/* get runtime PCM params using widget's stream name */
+	spcm = snd_sof_find_spcm_name(sdev, swidget->widget->sname);
+
 	/* process events */
 	switch (event) {
 	case SND_SOC_DAPM_PRE_PMU:
+		if (spcm->stream[stream].d0i3_active) {
```

Comment:
why would we test only for capture?

What if capture does not exist?

Reply:
Here the function sof_keyword_dapm_event is for KWD which is capture only, it existed in today's code like that 
```
line 153: ret = ipc_pcm_params(swidget, SOF_IPC_STREAM_CAPTURE);
```
So, we implement this event handler with assumption that there is and only is capture stream.